src/dotnet/commands/dotnet-restore/RestoreCommandParser.cs

```diff
@@ -31,7 +31,28 @@ private static Option[] FullRestoreOptions()
                         "--interactive",
                         LocalizableStrings.CmdInteractiveRestoreOptionDescription,
                         Accept.NoArguments()
-                              .ForwardAs("-property:NuGetInteractive=true")) }).ToArray();
+                            .ForwardAs("-property:NuGetInteractive=true")),
+                    Create.Option(
+                        "--use-lock-file",
```

Comment:
Is there a goal to make this feature on by default in a future (major) release? If so, the option might as well be a boolean switch defaulting to `true`, so one could also set `--use-lock-file false` (similar to `--self-contained false`)

Reply:
Currently there is no such plan to make it a default experience in dev16 so we'll continue with `--use-lock-file` switch for now. Even now there is another proposal `-ignore-lock-file` to be added which will allow achieving the similar behavior.